File: benchmark/c/cpw/write_you_own_power_without_using_multiplication_and_division/write_you_own_power_without_using_multiplication_and_division.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int write_you_own_power_without_using_multiplication_and_division ( int a, int b ) ;
int write_you_own_power_without_using_multiplication_and_division ( int a, int b ) {
  if ( b == 0 ) return 1;
  int answer = a;
  int increment = a;
  int i, j;
  for ( i = 1;
  i < b;
  i ++ ) {
    for ( j = 1;
    j < a;
    j ++ ) {
      answer += increment;
    }
    increment = answer;
  }
  return answer;
}

File: translation_result/else_bms/fail/c/cpw/write_you_own_power_without_using_multiplication_and_division/gpt4/Hinted/write_you_own_power_without_using_multiplication_and_division.rs

fn write_you_own_power_without_using_multiplication_and_division(a: i32, b: i32) -> i32 {
    if b == 0 {
        return 1;
    }
    let mut answer = a;
    let mut increment = a;
    
    for _ in 1..b {
        for _ in 1..a {
            answer = answer.wrapping_add(increment);
        }
        increment = answer;
    }
    
    answer
}
